dev to main (#1)
coretech/n2p-auth-service-client: README.md

> n2p-auth-service-client

Reviewer: **style (non blocking)** Use some proper markdown syntax to structure the readme file, e.g. headings

```suggestion
# net2phone Auth Service Client
```

> This is Auth Service Client project. This project allows any project to quickly consume Auth Server APIs in a .NET project.
> 
> Usage of the API We need to implement the logic to configure the service and use it with DI and other logic in the application.
> 
> Registering in DI, we require token provider implementation for the client to work.
> 
> Initialize : Firstly,you should initialize the auth service client. So, to configure the client we can use the next code

Reviewer: **nit (non blocking)**
```suggestion
This project allows any project to quickly consume Auth Server V2 APIs in a .NET project. To do so:

1. Implement a `IAuthServiceClientTokenProvider` interface to provide access token being used.
2. Configure necessary options.

```

>  using N2P.Auth.Service.Client;
>  public void ConfigureServices(IServiceCollection services)
>  
>  {
>      .....
>     services.AddAuthServiceApiClient<IAuthServiceClientTokenProvider>();
>  }
> 

Reviewer: **style (non blocking)**

Use code block styles to properly format code. Also, how the usage of options (and which options are available).

See https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax

```suggestion
```csharp
 using N2P.Auth.Service.Client;
 
 public void ConfigureServices(IServiceCollection services)
 {
    services.AddAuthServiceApiClient<CustomAuthServiceClientTokenProvider>(options => 
    {
      options.BaseUrl = "https://auth-qa.net2phone.com",
      options.Timeout = TimeSpan.FromSeconds(30)
    });
 }
```

```

API Clients package

Repository: https://github.com/coretech/n2p-auth-service-client
Jenkins build pipeline: --TODO Pipeline runs the logic to build the project and create the package, push the package to the JFrog Artifactory feed. As result of this, ** will be created the package: --TODO

> Integrate the package to the service/application When package created an deployed to the NuGet feed we can install the package from IDE (Visual Studio, Rider) or even added as reference source in the project file.

Reviewer: **suggestion (non blocking)** Let's add this when we have the CI and build pipeline. So for now I think we can remove this section.
